src/Oracle/Oracle.cs

```diff
@@ -0,0 +1,464 @@
+            OracleResponse response;
+            switch (uri.Scheme.ToLowerInvariant())
+            {
+                case "http":
+                case "https":
+                    try
+                    {
+                        var data = Https.Request(requestId, request.Url, request.Filter);
+                        response = new OracleResponse() { Id = requestId, Code = OracleResponseCode.Success, Result = data };
+                    }
+                    catch (FileNotFoundException notfoundex)
+                    {
+                        Log($"Request error {notfoundex.Message}");
+                        response = new OracleResponse() { Id = requestId, Code = OracleResponseCode.NotFound, Result = Array.Empty<byte>() };
+                    }
+                    catch (TimeoutException timeoutex)
+                    {
+                        Log($"Request error {timeoutex.Message}");
+                        response = new OracleResponse() { Id = requestId, Code = OracleResponseCode.Timeout, Result = Array.Empty<byte>() };
+                    }
+                    catch (Exception e)
+                    {
+                        Log($"Request error {e.Message}");
+                        response = new OracleResponse() { Id = requestId, Code = OracleResponseCode.Error, Result = Array.Empty<byte>() };
+                    }
+                    break;
+                default:
+                    Log($"{uri.Scheme.ToLowerInvariant()} is not supported");
+                    response = new OracleResponse() { Id = requestId, Code = OracleResponseCode.Forbidden, Result = Array.Empty<byte>() };
+                    break;
+            }
+            var responseTx = CreateResponseTx(snapshot, response);
+            var backupTx = CreateResponseTx(snapshot, new OracleResponse() { Code = OracleResponseCode.ConsensusUnreachable, Id = requestId, Result = Array.Empty<byte>() });
+
+            Log($"Builded response tx:{responseTx.Hash} requestTx:{request.OriginalTxid} requestId: {requestId}");
+
+            ECPoint[] oraclePublicKeys = NativeContract.Designate.GetDesignatedByRole(snapshot, Role.Oracle, snapshot.Height);
+            foreach (var account in Wallet.GetAccounts())
+            {
+                var oraclePub = account.GetKey().PublicKey;
+                if (!account.HasKey || account.Lock || !oraclePublicKeys.Contains(oraclePub)) continue;
+
+                var txSign = responseTx.Sign(account.GetKey());
+                var backTxSign = backupTx.Sign(account.GetKey());
+                AddResponseTxSign(snapshot, requestId, oraclePub, txSign, responseTx, backupTx, backTxSign);
+                SendResponseSignature(requestId, txSign, account.GetKey());
+
+                Log($"Send oracle sign data: Oracle node: {oraclePub} RequestTx: {request.OriginalTxid} Sign: {txSign.ToHexString()}");
+            }
+        }
+
+        private Transaction CreateResponseTx(StoreView snapshot, OracleResponse response)
```

Comment:
Any change we can make this public + static for use in neo-express?

Reply:
Done